agents/agents-features/agents-features-memory/src/commonMain/kotlin/ai/koog/agents/memory/feature/similarity/EmbeddingProvider.kt

```diff
@@ -0,0 +1,37 @@
+package ai.koog.agents.memory.feature.similarity
+
+/**
+ * Supplies vector representations of text for similarity comparisons.
+ *
+ * Implementations can wrap existing embedding infrastructure or provide lightweight in-memory behaviour.
+ */
+public interface EmbeddingProvider {
```

Comment:
Could you please elaborate why ` ai.koog.embeddings.base.Embedder` (and classes that implement it) doesn't fit here?